src/game_api/render_api.cpp

```diff
@@ -137,3 +139,49 @@ const char** RenderAPI::load_texture(std::string file_name)
     auto load_texture = (LoadTextureFunT*)get_load_texture();
     return load_texture(render_api, &file_name, 1);
 }
+
+using VanillaRenderHudFun = void(void*);
+VanillaRenderHudFun* g_render_hud_trampoline{nullptr};
+void render_hud(void* hud_data)
+{
+    g_render_hud_trampoline(hud_data);
+
+    LuaBackend::for_each_backend(
+        [&](LuaBackend& backend)
+        {
+            backend.vanilla_render();
+            return true;
+        });
+}
+
+void init_render_hud_hook()
+{
+    static size_t func_render_hud = 0;
+
+    auto& memory = Memory::get();
+    auto exe = memory.exe();
+
+    if (func_render_hud == 0)
+    {
+        std::string pattern = "\x48\x8D\x0D\x56\x05\x16\x00"s;

```

Comment:
I'm confused, what is the RVO of this function? Because if I use this pattern I find a function that def has more parameters than just one pointer.

Reply:
The function is at `221C1E90`. There's two extra parameters indeed, two floats for the y-level of the hud and its transparency. It doesn't really matter though, because the trampoline is the first function that gets called in the custom render_hud function, so `xmm1` and `xmm2` aren't touched and are passed on to the native function. I did add the params though, to be correct.
If you're looking at what Ghidra thinks the signature is, then as far as I can tell that's just wrong. If you look at where this function is called from, one level up in the call stack, you see `rcx`, `xmm1` and `xmm2` being prepared, that's all.

Reply:
I get your point, but imho it's better if we match the signature correctly. So in case we'll want to do anything else before calling the trampoline we don't suddenly break things without having an obvious reason.

And I know Ghidra is utterly useless with functions that take arguments in xmm registers, I got used to patching the signatures all the time  😬